### ISSUE TITLE
False Positive Error in Doc Test

### ISSUE THREAD

**Issue by ghost** [intellij-rust/intellij-rust#10779]
## Environment

* **IntelliJ Rust plugin version:** 0.4.199.5415-232
* **Rust toolchain version:** 1.71.0 (8ede3aae2 2023-07-12) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2023.2 Community Edition (IC-232.8660.185)
* **Operating system:** Windows 11.0
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.ide.intentions.macros.function-like, org.rust.mir.borrow-check, org.rust.mir.move-analysis, org.rust.macros.proc, org.rust.ssr

## Problem description
Rust allows for doc tests to elide the `fn main()`, even for doc tests which return a Result so long as the type is fully-qualified. IntelliJ Rust, however, flags these as errors. The example in the `Steps to reproduce` section compiles and runs (with `cargo test`) just fine, but generates this error in IntelliJ Rust:

```
mismatched types [E0308] 
expected `()`, found `Result<(), <unknown>>`
```

## Steps to reproduce
```rust
/// ```
/// # Ok::<(), ()>(())
/// ```
fn foo() {}
```

**Comment by Maksim20023** [comment]
Hi! Thank you for your report.
This is a known issue for us and has already been reported: see #8349. Unfortunately, we currently don't have information about ETA for this issue, and when we have any new information about it, we will add a reply to this ticket.
As a workaround, you can disable syntax highlighting for doctest.
To do so, please go to
`Settings | Languages & Frameworks | Rust`
and uncheck 
`Inject Rust language into documentation comments `
option.
Does it help?
